File: Python/Package/underautomation/universal_robots/internal/rtde_client_internal.py
import typing
from underautomation.universal_robots.rtde.rtde_output_setup import RtdeOutputSetup
from underautomation.universal_robots.rtde.rtde_input_setup import RtdeInputSetup
from underautomation.universal_robots.rtde.rtde_versions import RtdeVersions
from underautomation.universal_robots.rtde.internal.rtde_client_base import RtdeClientBase
import clr
import os
clr.AddReference(os.path.realpath(os.path.join(os.path.dirname(__file__), "..", "..", "..", 'lib', 'UnderAutomation.UniversalRobots.dll')))
from UnderAutomation.UniversalRobots.Internal import RtdeClientInternal as rtde_client_internal

class RtdeClientInternal(RtdeClientBase):
	def __init__(self, _internal = 0):
		if(_internal == 0):
			self._instance = rtde_client_internal()
		else:
			self._instance = _internal
	def connect(self, outputSetup: RtdeOutputSetup, inputSetup: RtdeInputSetup, version: RtdeVersions, frequency: float) -> None:
		self._instance.Connect(outputSetup._instance, inputSetup._instance, version._instance, frequency)

File: UnderAutomation.UniversalRobots.ObfuscatedSources/Internal/RtdeClientInternal.cs
//              WARNING
// This file is an empty shell containing only public C# items.
// The internal code is hidden; to access it, you need to obtain a Source licence of the library.

using Rtde.Internal;
using Rtde;

namespace Internal {
	/// <summary>
	/// Internal implementation of the Real-Time Data Exchange (RTDE) client that delegates connection to the parent <xref href="UnderAutomation.UniversalRobots.UR" data-throw-if-not-resolved="false"></xref> instance.
	/// </summary>
	public class RtdeClientInternal : RtdeClientBase {

		/// <summary>
		/// Connects to the RTDE interface on the robot controller.
		/// </summary>
		/// <param name="outputSetup">Configuration for RTDE output subscriptions.</param>
		/// <param name="inputSetup">Configuration for RTDE input registers.</param>
		/// <param name="version">RTDE protocol version to negotiate.</param>
		/// <param name="frequency">Output data frequency in Hz.</param>
		/// <param name="port">TCP port of the RTDE server.</param>
		public void Connect(RtdeOutputSetup outputSetup, RtdeInputSetup inputSetup, RtdeVersions version, double frequency, int port)
		{
			// Source is hidden, a Source licence is needed to access internal code...
		}
	}
}
